File: php/features.php
<?php
declare(strict_types=1);

// DungeonsAndDragonsTwo SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class DungeonsAndDragonsTwoFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new DungeonsAndDragonsTwoBaseFeature();
            case "test":
                return new DungeonsAndDragonsTwoTestFeature();
            default:
                return new DungeonsAndDragonsTwoBaseFeature();
        }
    }
}

File: rb/features.rb
# DungeonsAndDragonsTwo SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module DungeonsAndDragonsTwoFeatures
  def self.make_feature(name)
    case name
    when "base"
      DungeonsAndDragonsTwoBaseFeature.new
    when "test"
      DungeonsAndDragonsTwoTestFeature.new
    else
      DungeonsAndDragonsTwoBaseFeature.new
    end
  end
end
